osu.Game/Screens/Import/FileImportScreen.cs

```diff
@@ -160,5 +204,41 @@ private void startImport(string path)
                 });
             }, TaskCreationOptions.LongRunning);
         }
+
+        private void startDirectoryImport(string path)
+        {
+            if (string.IsNullOrEmpty(path))
+                return;
+
+            List<string> filesToImport = new List<string>();
+
+            try
+            {
+                foreach (string file in Directory.EnumerateFiles(path))
+                {
+                    // check if the file ends in a valid extension
+                    if (game.HandledExtensions.Contains(Path.GetExtension(file)))
+                        filesToImport.Add(file);
+                }
+            }
+            catch (Exception ex)
+            {
+                Logger.Error(ex, "Could not enumerate files in selected directory!");
+            }
+
+            if (filesToImport.Count <= 0)
+                return;
+
+            Task.Factory.StartNew(async () =>
+            {
+                await game.Import(filesToImport.ToArray()).ConfigureAwait(false);
+
+                // Refresh the filepicker after importing
+                Schedule(() =>
+                {
+                    fileSelector.CurrentPath.TriggerChange();
+                });
+            }, TaskCreationOptions.LongRunning);
```

Comment:
this is basically `startImport()` but inlined for some reason - just change the signature of `startImport()` to accept `params string[]` instead and use that here rather than copy-paste it here